### DISCUSSION TITLE
[Roadmap] Report building capabilities?

### DISCUSSION THREAD

**Discussion by ManZzup** [apache/superset#14685] [Ideas]
Hi all,

I've been evaluating Superset for a while now, and have in fact integrated it with a couple of products as the BI layer - absolutely love the features and extensibility of it. With the power of Superset's query engine, I'm wondering if there are any plans in the roadmap to create a complete report-building module on top of it. I'm aware of the current reporting options, but I'm thinking more around a fully formed report builder (like Birt or Jasper).

Thank you!

**Comment by hamin123** [comment]
Just submitted an issue along these lines. Did not realize, it was already part of discussion. Anyways, thanks for submitting. This is indeed a critical requirement for any open source BI tool.

**Comment by rvsoni** [comment]
I am looking to replace Pentaho BI (community) with Superset, The Reporting module is very important, 

We can filter the data in a dashboard, but as of now not able to schedule reports with the filters. 
So this is the most imp function to have to make a Superset a super reporting BI tool too.

**Reply by ManZzup** [reply]
@rvsoni as I'm aware, there's already a email report feature for dashboards and charts in the latest Superset version. We are also looking at creating something beyond that where a user can design a proper report as in Jasper while using Superset as the backend.

**Comment by ManZzup** [comment]
If anyone's following the discussion, we are currently implementing this capability on top of Superset. Hoping to release to the community soon and get engagement for maintainance.

**Reply by prateekj-bi** [reply]
Hi @ManZzup , Any updates on this. Did we have the functionality of creating Report Builder using Superset

**Reply by dbaltor** [reply]
I'm also interested in this feature. Can we have an update please?

**Reply by eschutho** [reply]
We can make sure to get some committers to look at your PR when you're ready. What features are you adding?

**Comment by rvsoni** [comment]
Thanks, @ManZzup for the details,

Jasper is very interesting, its Java reporting tool, and Superset is a Python project, I am not sure how both are going to work together, 

Designing Jaser report is also a development task, need to know the Jesper report template and its design tools. 

At this point, if we can set the filter as we do with dashboard, would make my day to export data with schedule reports and with a set of filter values and data range, 

Thanks,
Ravi

**Comment by eschutho** [comment]
Some folks from Preset are currently looking at adding PDF reports in the next few months, and there is also a [discussion](https://github.com/apache/superset/discussions/24843) from 
[@Ville Brofeldt](https://apache-superset.slack.com/team/UH0UPCJVD)
 about permalinks to use with filtering reports.

**Comment by aangarkar** [comment]
@ManZzup We were able to customise superset frontend to address this "report builder" functionality to the extend we can live with it for now. We should definitely look at make this happen within Superset as default.

**Comment by ameenhashir** [comment]
Are there any update on Report Designer implementation with Superset?
Am looking for a proper report with a template design and download in different file formats like pdf,excel,csv etc
